src/error-table.rkt

```diff
@@ -1,20 +1,365 @@
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
+                   
+                   #|
+                   TODO Make this actually work
+                   |#
+                   [(list 'sin.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val  (flabs (hash-ref exacts-hash arg)))
+                    (and (> arg-val 1e30) (mark-erroneous! subexpr pt))]
+                   
+                   [(list 'cos.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val  (flabs (hash-ref exacts-hash arg)))
+                    (and (> arg-val 1e30) (mark-erroneous! subexpr pt))]
+                   
+                   [(list 'tan.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val (flabs (hash-ref exacts-hash arg)))
+                    (and (> arg-val 1e30) (mark-erroneous! subexpr pt))]
+                   
+                   [(list 'sqrt.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val (hash-ref exacts-hash arg))
+
+                    ; Under/overflow rescue:
+                    (and (or (underflow? arg)
+                             (overflow? arg))
+                         (not (= subexpr-val arg-val))
+                         (mark-erroneous! subexpr pt))]
+
+                   [(list 'cbrt.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val (hash-ref exacts-hash arg))
+                    
+                    ; Under/overflow rescue:
+                    (and (or (underflow? arg)
+                             (overflow? arg))
+                         (not (= subexpr-val arg-val))
+                         (mark-erroneous! subexpr pt))]
+                   
+                   [(list '/.f64 x-ex y-ex)
+                    #:when (or (is-inexact? x-ex) (is-inexact? y-ex))
+                    (define x (hash-ref exacts-hash x-ex))
+                    (define y (hash-ref exacts-hash y-ex))
+                    (define x/y (/ x y))
+                    (define y-oflow? (overflow? y-ex))
+                    (define x-uflow? (underflow? x-ex))
+                    (define x-oflow? (overflow? x-ex))
+                    (define y-uflow? (underflow? y-ex))
+                    
+                    (cond
+                      
+                      ; Why am I doing this?
+                      [(and (= x/y 0.0) (underflow? subexpr)) #f]
+                      
+                      ; Why am I doing this?
+                      [(and (infinite? x/y) (overflow? subexpr)) #f]
+                      
+                      ; x underflows and y rescues it
+                      [(and x-uflow? (not (= subexpr-val x)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; y underflows and x rescues it
+                      [(and y-uflow? (not (nan? subexpr-val)))
```

Comment:
What if y underflows but correct answer is infinity?

Reply:
The conditions should be more clearer now